File: apps/askbible-android/app/src/main/java/me/askbible/native_/ui/PostReadingEditions.kt
package me.askbible.native_.ui

import me.askbible.native_.data.SiteCopy
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.defaultMinSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.layout.layout
import androidx.compose.ui.platform.LocalConfiguration
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import me.askbible.native_.data.InfoEditionDatabase
import me.askbible.native_.data.InfoEditionFormat
import me.askbible.native_.data.InfoEditionVariant
import me.askbible.native_.data.Parchment
import me.askbible.native_.data.ReadSize

private val LEAD = Color(0xE6784B1E)
private val HINT = Color(0xEB8C5A2A)
private val NAV_INK = Color(0xFF8C5A2A)
private val PAPER = Color(0xFFF2E4CF)

/**
 * 章末「读后两版」入口。对应 RN ReadChapterPostReadingEditions + PostReadingBookPage + ReadChapterInfoEditionBlock，与 iOS 的 PostReadingEditions 对等：
 * 标题「继续阅读与思考」+ 引导语 + 细线；左右两页书脊卡「陪你探索 / 查找资料」；点开后在下方铺出该版正文（纸面卡片），
 * 末尾「返回」；再往下是「上一章 / 回到顶部 / 下一章」。字号随阅读档位 textScale = verseFontSize / 16。
 */
@Composable
fun PostReadingEditions(
    bookId: String,
    chapter: Int,
    size: ReadSize,
    theme: Parchment,
    prev: ChapterNeighbor?,
    next: ChapterNeighbor?,
    active: InfoEditionVariant?,
    /** 用英文那套讲解 / 发现（读英文译本，或界面是英文时） */
    english: Boolean = false,
    onActiveChange: (InfoEditionVariant?) -> Unit,
    onNavigate: (String, Int) -> Unit,
    onBackToTop: () -> Unit,
) {
    val scale = (size.metrics.verseFontSize / 16f).coerceIn(0.8f, 2.8f)
    fun sx(n: Float) = Math.round(n * scale * 10f) / 10f

    Column(Modifier.fillMaxWidth().padding(top = 32.dp), horizontalAlignment = Alignment.CenterHorizontally) {
        // heading（下 20）
        Text(SiteCopy.t("pages.read.postReadingEditionsHeading"), color = theme.ink.toColor(), fontSize = sx(22f).sp, lineHeight = sx(30f).sp,
             fontWeight = FontWeight.SemiBold, letterSpacing = 0.8.sp, textAlign = TextAlign.Center,
             modifier = Modifier.padding(bottom = 10.dp))
        Text(SiteCopy.t("pages.read.postReadingEditionsLead"), color = LEAD, fontSize = sx(13f).sp, lineHeight = sx(21f).sp,
             textAlign = TextAlign.Center, modifier = Modifier.padding(horizontal = 18.dp).padding(bottom = 10.dp))
        Text(SiteCopy.t("pages.read.postReadingEditionsTapHint"), color = HINT, fontSize = sx(12f).sp, lineHeight = sx(18f).sp, fontWeight = FontWeight.Medium,
             letterSpacing = 0.2.sp, textAlign = TextAlign.Center, modifier = Modifier.padding(bottom = 8.dp))
        BoxWithConstraints(Modifier.fillMaxWidth().padding(bottom = 20.dp), contentAlignment = Alignment.Center) {
            Box(Modifier.width(minOf(224.dp, maxWidth * 0.56f)).height(0.5.dp).background(Color(0x47483422)))
        }

        // bookSpread：两页各占一半，外圈圆角 12；页面本身透明（羊皮透出）
        Row(Modifier.fillMaxWidth().clip(RoundedCornerShape(12.dp)), verticalAlignment = Alignment.Top) {
            Page(InfoEditionVariant.GUIDE, active == InfoEditionVariant.GUIDE, ::sx, Modifier.weight(1f)) { onActiveChange(InfoEditionVariant.GUIDE) }
            Page(InfoEditionVariant.INFO, active == InfoEditionVariant.INFO, ::sx, Modifier.weight(1f)) { onActiveChange(InfoEditionVariant.INFO) }
        }

        if (active != null) {
            EditionBlock(bookId, chapter, active, english, size, theme, ::sx, onBack = { onActiveChange(null) }, onLink = { url ->
                (ArticleLink.resolve(url) as? ArticleLink.Chapter)?.let { onNavigate(it.bookId, it.chapter) }
            })
            // 底部「上一章 / 回到顶部 / 下一章」（上下各 50）
            Row(Modifier.fillMaxWidth().padding(top = 50.dp, bottom = 50.dp), verticalAlignment = Alignment.CenterVertically) {
                Box(Modifier.weight(1f), contentAlignment = Alignment.CenterStart) {
                    prev?.let { p ->
                        Row(Modifier.clickableNoRipple { onNavigate(p.bookId, p.chapter) }.padding(6.dp),
                            verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(1.dp)) {
                            MaterialIcon(MI.CHEVRON_LEFT, 16f, NAV_INK)
                            Text(SiteCopy.t("pages.read.chapterEndNavPrev"), color = Color(0xE08C5A2A), fontSize = 13.sp, fontWeight = FontWeight.Medium, letterSpacing = 0.1.sp)
                        }
                    }
                }
                Text(SiteCopy.t("pages.read.postReadingBackToTop"), color = Color(0xD68C5A2A), fontSize = sx(14f).sp, fontWeight = FontWeight.Medium, letterSpacing = 0.2.sp,
                     modifier = Modifier.clickableNoRipple(onBackToTop).padding(horizontal = 8.dp, vertical = 6.dp))
                Box(Modifier.weight(1f), contentAlignment = Alignment.CenterEnd) {
                    next?.let { n ->
                        Row(Modifier.clickableNoRipple { onNavigate(n.bookId, n.chapter) }.padding(6.dp),
                            verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(1.dp)) {
                            Text(SiteCopy.t("pages.read.chapterChromeNext"), color = Color(0xE08C5A2A), fontSize = 13.sp, fontWeight = FontWeight.Medium, letterSpacing = 0.1.sp)
                            MaterialIcon(MI.CHEVRON_RIGHT, 16f, NAV_INK)
                        }
                    }
                }
            }
        }
    }
}

/** PostReadingBookPage：方形插画（stretch，80%）+ 标题 17/600 强调色 + 简介 11 + 「点按打开 ›」/「已选择 ✓」 */
@Composable
private fun Page(variant: InfoEditionVariant, isActive: Boolean, sx: (Float) -> Float, modifier: Modifier, onPress: () -> Unit) {
    val art = rememberAssetImage(if (variant == InfoEditionVariant.GUIDE) "images/post-reading-discover.png" else "images/post-reading-consult.png")
    val title = SiteCopy.t(if (variant == InfoEditionVariant.GUIDE) "pages.read.postReadingEditionGuideTitle" else "pages.read.postReadingEditionInfoTitle")
    val blurb = SiteCopy.t(if (variant == InfoEditionVariant.GUIDE) "pages.read.postReadingEditionGuideBlurb" else "pages.read.postReadingEditionInfoBlurb")
    Column(modifier.clickableNoRipple { if (!isActive) onPress() }, horizontalAlignment = Alignment.CenterHorizontally) {
        if (art != null) {
            Image(art, contentDescription = null, contentScale = ContentScale.FillBounds,
                  modifier = Modifier.fillMaxWidth().aspectRatio(1f).alpha(0.8f))
        } else {
            Box(Modifier.fillMaxWidth().aspectRatio(1f))
        }
        Column(Modifier.fillMaxWidth().padding(start = 12.dp, end = 12.dp, top = 4.dp, bottom = 14.dp),
               horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(7.dp)) {
            Text(title, color = MARKDOWN_ACCENT, fontSize = sx(17f).sp, lineHeight = sx(24f).sp, fontWeight = FontWeight.SemiBold,
                 letterSpacing = 0.6.sp, textAlign = TextAlign.Center)
            Text(blurb, color = Color(0xDB784B1E), fontSize = sx(11f).sp, lineHeight = sx(17f).sp, textAlign = TextAlign.Center,
                 modifier = Modifier.widthIn(max = 168.dp))
            Row(Modifier.padding(top = 2.dp).defaultMinSize(minHeight = 18.dp), verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(2.dp)) {
                Text(SiteCopy.t(if (isActive) "pages.read.postReadingEditionSelected" else "pages.read.postReadingEditionTapAction"), color = HINT, fontSize = 11.sp, lineHeight = 16.sp,
                     fontWeight = FontWeight.Medium, letterSpacing = 0.2.sp)
                MaterialIcon(if (isActive) MI.CHECK_CIRCLE else MI.CHEVRON_RIGHT, 14f, if (isActive) Color(0xFF7A633A) else NAV_INK)
            }
        }
    }
}

/**
 * 展开的一版正文。对应 RN ReadChapterInfoEditionBlock：免责声明 → 通屏壳（顶部一道 15dp 的暗影）→
 * 纸面卡片（#F2E4CF、圆角 18、边 rgba(150,112,64,.18)、投影）→ 标题 + Markdown → 「返回」。
 */
@Composable
private fun EditionBlock(
    bookId: String, chapter: Int, variant: InfoEditionVariant, english: Boolean, size: ReadSize, theme: Parchment,
    sx: (Float) -> Float, onBack: () -> Unit, onLink: (String) -> Unit,
) {
    val context = LocalContext.current
    val content = remember(bookId, chapter, variant, english) {
        InfoEditionDatabase.open(context)?.chapter(bookId, chapter, variant, english)?.let { ch ->
            InfoEditionFormat.splitPrimaryHeading(InfoEditionFormat.readerText(ch.markdown, variant))
        }
    }
    val screenH = LocalConfiguration.current.screenHeightDp.dp
    Column(Modifier.fillMaxWidth().padding(top = 24.dp), horizontalAlignment = Alignment.CenterHorizontally) {
        Text(SiteCopy.t("pages.read.infoEditionDisclaimer"), color = theme.muted.toColor(), fontSize = sx(12f).sp, lineHeight = sx(19f).sp,
             textAlign = TextAlign.Center, modifier = Modifier.widthIn(max = 320.dp).padding(horizontal = 8.dp).padding(top = 10.dp, bottom = 18.dp))

        // bodyFullscreenShell：通屏（越过章页 20 的内边距），最小高度一屏，顶部一道暗影
        val shape = RoundedCornerShape(18.dp)
        Box(
            Modifier.fillMaxWidth()
                .layout { measurable, constraints ->
                    val extra = 40.dp.roundToPx()
                    val p = measurable.measure(constraints.copy(maxWidth = constraints.maxWidth + extra, minWidth = constraints.maxWidth + extra))
                    layout(constraints.maxWidth, p.height) { p.placeRelative(-extra / 2, 0) }
                }
                .heightIn(min = screenH)
        ) {
            Box(Modifier.fillMaxWidth().height(15.dp).background(Brush.verticalGradient(
                0f to Color(0x7A2A180D), 0.45f to Color(0x4F1D120A), 1f to Color.Transparent)))
            Column(
                Modifier.fillMaxWidth().padding(start = 12.dp, end = 12.dp, top = 14.dp, bottom = 14.dp)
                    .shadow(16.dp, shape, ambientColor = Color(0x38000000), spotColor = Color(0x38000000))
                    .clip(shape).background(PAPER).border(1.dp, Color(0x2E967040), shape)
                    .padding(start = 18.dp, end = 18.dp, top = 20.dp, bottom = 22.dp),
            ) {
                if (content != null) {
                    content.first?.let { h ->
                        // titleStyles：24/700 强调色、行高 36、字距 .5、居中、上 30 下 22
                        Text(h, Modifier.fillMaxWidth().padding(top = sx(30f).dp, bottom = 22.dp), color = MARKDOWN_ACCENT,
                             fontSize = sx(24f).sp, lineHeight = sx(36f).sp, fontWeight = FontWeight.Bold, letterSpacing = 0.5.sp,
                             textAlign = TextAlign.Center)
                    }
                    MarkdownBody(content.second, size, theme, onLink)
                } else {
                    Text(SiteCopy.t("pages.read.infoEditionLoadFailed"), Modifier.fillMaxWidth(), color = theme.muted.toColor(), fontSize = sx(13f).sp,
                         lineHeight = sx(20f).sp, textAlign = TextAlign.Center)
                }
                Box(Modifier.fillMaxWidth().padding(top = 50.dp, bottom = 100.dp), contentAlignment = Alignment.Center) {
                    Text(SiteCopy.t("pages.read.chapterChromeBack"), color = NAV_INK, fontSize = sx(14f).sp, fontWeight = FontWeight.SemiBold, letterSpacing = 0.3.sp,
                         modifier = Modifier.clickableNoRipple(onBack).padding(horizontal = 8.dp, vertical = 6.dp))
                }
            }
        }
    }
}

File: apps/askbible-ios/AskBible/Read/PostReadingEditions.swift
import SwiftUI

/// 章末「读后两版」入口。对应 RN ReadChapterPostReadingEditions + PostReadingBookPage + ReadChapterInfoEditionBlock：
/// 标题「继续阅读与思考」+ 引导语 + 细线；左右两页书脊卡「陪你探索 / 查找资料」；点开后在下方铺出该版正文（纸面卡片），
/// 末尾「返回」；再往下是「上一章 / 回到顶部 / 下一章」。字号随阅读档位 textScale = verseFontSize / 16。
struct PostReadingEditions: View {
    let bookId: String
    let chapter: Int
    var size: ReadSize = .default
    let theme: Parchment
    var prev: ChapterNeighbor?
    var next: ChapterNeighbor?
    var onNavigate: (_ bookId: String, _ chapter: Int) -> Void
    var onBackToTop: () -> Void
    @Binding var active: InfoEditionVariant?

    private var scale: CGFloat { max(0.8, min(2.8, size.metrics.verseFontSize / 16)) }
    private func sx(_ n: CGFloat) -> CGFloat { (n * scale * 10).rounded() / 10 }

    private static let lead = Color(red: 120 / 255, green: 75 / 255, blue: 30 / 255, opacity: 0.9)
    private static let hint = Color(red: 140 / 255, green: 90 / 255, blue: 42 / 255, opacity: 0.92)
    private static let navInk = Color(rgb: 0x8C5A2A)
    private static let paper = Color(rgb: 0xF2E4CF)

    var body: some View {
        VStack(spacing: 0) {
            heading
            bookSpread
            if let v = active {
                EditionBlock(bookId: bookId, chapter: chapter, variant: v, size: size, theme: theme, onBack: { active = nil })
                bottomRow
            }
        }
        .padding(.top, 32)
    }

    private var heading: some View {
        VStack(spacing: 0) {
            Text("继续阅读与思考")
                .font(.system(size: sx(22), weight: .semibold)).tracking(0.8)
                .lineSpacing(max(0, sx(30) - sx(22)))
                .foregroundStyle(theme.ink).multilineTextAlignment(.center)
                .padding(.bottom, 10)
            Text("你可以先安静查阅资料，再回到经文里。")
                .font(.system(size: sx(13))).lineSpacing(max(0, sx(21) - sx(13)))
                .foregroundStyle(Self.lead).multilineTextAlignment(.center)
                .padding(.horizontal, 18).padding(.bottom, 10)
            Text("点按下方任一卡片进入")
                .font(.system(size: sx(12), weight: .medium)).tracking(0.2)
                .lineSpacing(max(0, sx(18) - sx(12)))
                .foregroundStyle(Self.hint).multilineTextAlignment(.center)
                .padding(.bottom, 8)
            GeometryReader { geo in
                Rectangle().fill(Color(red: 72 / 255, green: 52 / 255, blue: 34 / 255, opacity: 0.28))
                    .frame(width: min(224, geo.size.width * 0.56), height: 1 / UIScreen.main.scale)
                    .frame(maxWidth: .infinity)
            }
            .frame(height: 1)
        }
        .padding(.bottom, 20)
    }

    /// bookSpread：两页各占一半，外圈圆角 12；页面本身透明（羊皮透出）
    private var bookSpread: some View {
        HStack(alignment: .top, spacing: 0) {
            page(.guide)
            page(.info)
        }
        .clipShape(RoundedRectangle(cornerRadius: 12))
    }

    private func page(_ v: InfoEditionVariant) -> some View {
        let isActive = active == v
        let art = v == .guide ? "post-reading-discover" : "post-reading-consult"
        let title = v == .guide ? "陪你探索" : "查找资料"
        let blurb = v == .guide ? "通过问题引导去探索" : "整理资料供你参考"
        return Button {
            if active != v { active = v }
        } label: {
            VStack(spacing: 0) {
                if let img = BundleImage.load(art, ext: "png") {
                    Image(uiImage: img).resizable().aspectRatio(1, contentMode: .fill).opacity(0.8)
                } else {
                    Color.clear.aspectRatio(1, contentMode: .fill)
                }
                VStack(spacing: 7) {
                    Text(title)
                        .font(.system(size: sx(17), weight: .semibold)).tracking(0.6)
                        .lineSpacing(max(0, sx(24) - sx(17)))
                        .foregroundStyle(MarkdownBody.accent).multilineTextAlignment(.center)
                    Text(blurb)
                        .font(.system(size: sx(11))).lineSpacing(max(0, sx(17) - sx(11)))
                        .foregroundStyle(Color(red: 120 / 255, green: 75 / 255, blue: 30 / 255, opacity: 0.86))
                        .multilineTextAlignment(.center).frame(maxWidth: 168)
                    HStack(spacing: 2) {
                        Text(isActive ? "已选择" : "点按打开")
                            .font(.system(size: 11, weight: .medium)).tracking(0.2)
                            .foregroundStyle(Self.hint)
                        MaterialIcon(glyph: isActive ? MI.checkCircle : MI.chevronRight, size: 14,
                                     color: isActive ? Color(rgb: 0x7A633A) : Self.navInk)
                    }
                    .frame(minHeight: 18).padding(.top, 2)
                }
                .padding(.top, 4).padding(.horizontal, 12).padding(.bottom, 14)
                .frame(maxWidth: .infinity)
            }
        }
        .buttonStyle(.plain)
        .frame(maxWidth: .infinity)
    }

    /// 底部「上一章 / 回到顶部 / 下一章」（上下各 50）
    private var bottomRow: some View {
        HStack(spacing: 0) {
            HStack {
                if let p = prev {
                    Button { onNavigate(p.bookId, p.chapter) } label: {
                        HStack(spacing: 1) {
                            MaterialIcon(glyph: MI.chevronLeft, size: 16, color: Self.navInk)
                            Text("上一章").font(.system(size: 13, weight: .medium)).tracking(0.1)
                                .foregroundStyle(Color(red: 140 / 255, green: 90 / 255, blue: 42 / 255, opacity: 0.88))
                        }
                        .padding(6)
                    }
                    .buttonStyle(.plain)
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            Button(action: onBackToTop) {
                Text("回到顶部").font(.system(size: sx(14), weight: .medium)).tracking(0.2)
                    .foregroundStyle(Color(red: 140 / 255, green: 90 / 255, blue: 42 / 255, opacity: 0.84))
                    .padding(.horizontal, 8).padding(.vertical, 6)
            }
            .buttonStyle(.plain)
            HStack {
                if let n = next {
                    Button { onNavigate(n.bookId, n.chapter) } label: {
                        HStack(spacing: 1) {
                            Text("下一章").font(.system(size: 13, weight: .medium)).tracking(0.1)
                                .foregroundStyle(Color(red: 140 / 255, green: 90 / 255, blue: 42 / 255, opacity: 0.88))
                            MaterialIcon(glyph: MI.chevronRight, size: 16, color: Self.navInk)
                        }
                        .padding(6)
                    }
                    .buttonStyle(.plain)
                }
            }
            .frame(maxWidth: .infinity, alignment: .trailing)
        }
        .padding(.top, 50).padding(.bottom, 50)
    }
}

/// 展开的一版正文。对应 RN ReadChapterInfoEditionBlock：免责声明 → 通屏壳（顶部一道 15pt 的暗影）→
/// 纸面卡片（#F2E4CF、圆角 18、边 rgba(150,112,64,.18)、投影）→ 标题 + Markdown → 「返回」。
private struct EditionBlock: View {
    let bookId: String
    let chapter: Int
    let variant: InfoEditionVariant
    var size: ReadSize
    let theme: Parchment
    var onBack: () -> Void

    private var scale: CGFloat { max(0.8, min(2.8, size.metrics.verseFontSize / 16)) }
    private func sx(_ n: CGFloat) -> CGFloat { (n * scale * 10).rounded() / 10 }

    private var loaded: (heading: String?, body: String)? {
        guard let ch = InfoEditionDatabase.shared?.chapter(bookId: bookId, chapter: chapter, variant: variant) else { return nil }
        return InfoEditionFormat.splitPrimaryHeading(InfoEditionFormat.readerText(ch.markdown, variant: variant))
    }

    var body: some View {
        let content = loaded
        VStack(spacing: 0) {
            Text("以下仅为参考资料，请对照圣经慎思明辨")
                .font(.system(size: sx(12))).lineSpacing(max(0, sx(19) - sx(12)))
                .foregroundStyle(theme.muted).multilineTextAlignment(.center)
                .frame(maxWidth: 320).padding(.horizontal, 8)
                .padding(.top, 10).padding(.bottom, 14 + 4)

            VStack(alignment: .leading, spacing: 0) {
                if let content {
                    if let h = content.heading {
                        // titleStyles：24/700 强调色、行高 36、字距 .5、居中、上 30 下 22
                        Text(h)
                            .font(.system(size: sx(24), weight: .bold)).tracking(0.5)
                            .lineSpacing(max(0, sx(36) - sx(24)))
                            .foregroundStyle(MarkdownBody.accent).multilineTextAlignment(.center)
                            .frame(maxWidth: .infinity)
                            .padding(.top, sx(30)).padding(.bottom, 22)
                    }
                    MarkdownBody(markdown: content.body, size: size, theme: theme)
                } else {
                    Text("暂时无法加载本章讲解")
                        .font(.system(size: sx(13))).lineSpacing(max(0, sx(20) - sx(13)))
                        .foregroundStyle(theme.muted).frame(maxWidth: .infinity)
                }
                Button(action: onBack) {
                    Text("返回").font(.system(size: sx(14), weight: .semibold)).tracking(0.3)
                        .foregroundStyle(Color(rgb: 0x8C5A2A))
                        .padding(.horizontal, 8).padding(.vertical, 6)
                }
                .buttonStyle(.plain)
                .frame(maxWidth: .infinity)
                .padding(.top, 50).padding(.bottom, 100)
            }
            .padding(.horizontal, 18).padding(.top, 20).padding(.bottom, 22)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(RoundedRectangle(cornerRadius: 18).fill(Color(rgb: 0xF2E4CF)))
            .overlay(RoundedRectangle(cornerRadius: 18).strokeBorder(Color(red: 150 / 255, green: 112 / 255, blue: 64 / 255, opacity: 0.18), lineWidth: 1))
            .shadow(color: .black.opacity(0.22), radius: 16, y: 8)
            .padding(.horizontal, 12).padding(.top, 14).padding(.bottom, 14)
            .frame(maxWidth: .infinity, minHeight: UIScreen.main.bounds.height, alignment: .top)
            .background(alignment: .top) {
                // bodyFullscreenShell 顶部的暗影：rgba(42,24,13,.48) → 15pt 内淡出
                LinearGradient(stops: [
                    .init(color: Color(red: 42 / 255, green: 24 / 255, blue: 13 / 255, opacity: 0.48), location: 0),
                    .init(color: Color(red: 29 / 255, green: 18 / 255, blue: 10 / 255, opacity: 0.31), location: 0.45),
                    .init(color: .clear, location: 1),
                ], startPoint: .top, endPoint: .bottom)
                .frame(height: 15)
            }
            .padding(.horizontal, -20)  // 通屏：越过章页 20 的内边距
        }
        .padding(.top, 24)
    }
}
